File: Week-1/Quick-Union/QuickUnionUF.java
public class QuickUnionUF
    {
        private int[] id;
        
        
        // Set id of each object to itself
        public QuickUnionUF(int N) {
            id = new int[N];
            for (int i = 0; i < N; i++) {
                id[i] = i; //set id of each object to itself
            }
        }
        
        // Chase parent pointers until reach root
        private int root(int i) {
            while (i != id[i]) {
                i = id[i];
            }
            
            return i;
        }

        // Check if p and q have same root
        public boolean connected(int p, int q) {
            return root(p) == root(q);
        }
        
        // Change root of p to point to root of q
        public void union(int p, int q) {
            int i = root(p);
            int j = root(q);
            id[i] = j;
        }
    }

File: Week-1/Quick-Union/QuickUnionUF.swift
class QuickUnionUF {
    //Data structure
    var id: [Int]
    
    //Set id of each object to itself
    init(N: Int)
    {
        id = [Int]()
        for i in 0...N {
            id[i] = i
        }
    }
    
    //Chase parent pointers until reach root
    func root(var i: Int) -> Int {
        while i != id[i] {
            i = id[i];
        }
        return i;
    }
    //Check if p and q have the same root
    func connected(p: Int, q: Int) -> Bool {
        return root(p) == root(q)
    }
    
    //Change root of p to point to root of q
    func union(p: Int, q: Int) {
        let i = root(p)
        let j = root(q)
        id[i] = j
    }
    
}
